templates/uefi/generic-qemu-uefi-boot-template.jinja2

```diff
@@ -0,0 +1,62 @@
+{% extends 'base/kernel-ci-base.jinja2' %}
+{% set ctx_arch = arch %}
```

Comment:
This is very similar to `generic-qemu-boot-template.jinja2`, so I think it should be inheriting from it to avoid duplicating all this.  Then you can add the i386 case in the base template.